torch/csrc/api/include/torch/enum.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include <string>
+
+#include <torch/csrc/WindowsTorchApiMacro.h>
+
+#define TORCH_ENUM_DECLARE(name) \
+namespace torch { \
+namespace enumtype { \
```

Comment:
How'd you decide on this name? Is it just an internal name or is it visible to users?

Reply:
It is only an internal name - the type of the enum is e.g. `torch::enumtype::kFanIn`, and the variable that has this type is `torch::kFanIn`, which is what's used by the end user.